### PR TITLE
Fix/321 members are not deleted even though the 204 response is returned

### LINKED ISSUE #321: Members are not deleted even though the 204 response is returned
## Bug Description

Members are not deleted even though the 204 response is returned. This issue is observed on the frontend, but it should stem from the backend.

## Steps to reproduce the bug

1. Navigate to the `env.development.local` file in the frontend repository on your local machine.
2. In this file, specify the backend API endpoint as the one for the AWS development environment.
3. Launch the local server for the frontend by running the command `npm run dev` and visit the local app in your browser.
4. Attempt to delete a member of a group while using multiple accounts for testing purposes.
5. Even though you will receive a 204 response through the API call, the user you attempted to delete will still be present.

## Expected Behavior

The user should be deleted after receiving a 204 response.

## Screen Capture

In the video, although the page is refreshed after attempting to delete the user, it still persists.

https://github.com/nishiki-tech/nishiki-backend/assets/99148565/47557650-c14f-4ab6-92cc-a85d734d980b

### PR DESCRIPTION
**Pull request by taniguchiiqqq**
## Overviews of implementation
Fix the update method of GroupRepository so that it can delete users who don't exist in a udpated group.

## Review points
In the future, when we develop the new feature to a container to another group, it need to be modified. But for now, I leave it.

### CODE REVIEW

File: backend/main/src/Group/Repositories/GroupRepository.ts [hunk 1]
```diff
@@ -79,11 +79,24 @@ export class GroupRepository implements IGroupRepository {
...
+			Group.id.id,
+		);
+		const deletingUsers = currentGroupUsers.filter(
+			(user) => !Group.userIds.some((userId) => userId.id === user.userId),
+		);
+
+		await Promise.all([
+			deletingUsers.map((user) =>
+				this.nishikiDbClient.deleteUserFromGroup(Group.id.id, user.userId),
+			),
```
Comment (hitohata) [inline backend/main/src/Group/Repositories/GroupRepository.ts:101]: I'm not sure, don't need it to spread it?

File: backend/main/src/Group/Repositories/GroupRepository.ts
Comment (hitohata) [inline backend/main/src/Group/Repositories/GroupRepository.ts:1]: I think this operation also needs to implement the delete container operation.

**Review by hitohata — COMMENTED** [review]
Thank you for your implementation.
Could you check the comments?

File: backend/main/src/Group/Repositories/GroupRepository.ts (hunk 1 above)
Comment (taniguchiiqqq) [reply backend/main/src/Group/Repositories/GroupRepository.ts:101]: Sorry, I don't get it. But at least I ensured that it worked in aws.

File: backend/main/src/Group/Repositories/GroupRepository.ts
Comment (taniguchiiqqq) [reply backend/main/src/Group/Repositories/GroupRepository.ts:1]: Thank you so much. I overlooked that there is data I need to delete explicitly. I was just thinking that the GSI between a container and a group is automatically deleted. I fixed it.

**Comment by taniguchiiqqq** [comment]
@hitohata 
Sorry, I couldn't figure out why chech-type-document test failed. If you have any idea, please let me know.

**Comment by hitohata** [comment]
> @hitohata Sorry, I couldn't figure out why chech-type-document test failed. If you have any idea, please let me know.

I have no idea but it seems errors have occurred continually, and the GitHub Actions doesn't raise any errors.
https://github.com/nishiki-tech/nishiki-backend/actions/runs/8349537124/job/22853937550#step:3:51

Anyway, I have fixed it. So, please check and merge the following PR: 
https://github.com/nishiki-tech/nishiki-backend/pull/328

**Comment by taniguchiiqqq** [comment]
@hitohata 
Thank you so much. It's weird behavior, but for now I'll approve your fix PR

**Review by hitohata — APPROVED** [review]
Thank you. LGTM.

**Pull request merged**